File: Algoritma/Pengurutan/Bubble Sort/BubbleSort.java
import java.util.Scanner;
import java.util.Random;

public class BubbleSort
{
    static Scanner scan = new Scanner (System.in);
    static Random random = new Random ();

    public static void main (String [] args)
    {
        int pilihan = 0, ukuran = 0;

        do
        {
            System.out.println ("========== BUBBLE SORT ==========");
            System.out.println ("1. Isi array secara manual");
            System.out.println ("2. Isi array secara otomatis (acak)");
            System.out.println ("3. Keluar");
            System.out.print ("Pilih salah satu: ");
            pilihan = scan.nextInt ();

            switch (pilihan)
            {
                case 1: manual (ukuran);
                        break;
                case 2: otomatis (ukuran);
                        break;
                case 3: System.out.println ("Terima kasih telah menggunakan program ini!");
                        break;
                default: System.out.println ("Input error");
            }
        } while (pilihan != 3);

        scan.close ();
    }

    static void manual (int ukuran)
    {
        System.out.print ("Masukkan batas ukuran array: ");
        ukuran = scan.nextInt ();

        int [] array = new int [ukuran];

        System.out.println ("Masukkan nilai tiap elemen array:");
        for (int i = 0; i < ukuran; i++)
        {
            array [i] = scan.nextInt ();
        }

        urut_dan_tampilkan (ukuran, array);
    }

    static void otomatis (int ukuran)
    {
        int batas = 0;

        System.out.print ("Masukkan batas ukuran array: ");
        ukuran = scan.nextInt ();
        System.out.print ("Masukkan batas angka acak: ");
        batas = scan.nextInt ();

        int [] array = new int [ukuran];

        for (int i = 0; i < ukuran; i++)
        {
            array [i] = random.nextInt (batas) + 1;
        }

        System.out.println ("Elemen array:");
        for (int i = 0; i < ukuran; i++)
        {
            System.out.print (array [i] + " ");
        }
        System.out.println ();

        urut_dan_tampilkan (ukuran, array);
    }

    static void urut_dan_tampilkan (int ukuran, int [] array)
    {
        for (int i = 0; i < ukuran - 1; i++)
        {
            for (int j = 0; j < ukuran - i - 1; j++)
            {
                if (array [j] > array [j + 1])
                {
                    int sementara = array [j];
                    array [j] = array [j + 1];
                    array [j + 1] = sementara;
                }
            }
        }

        System.out.println ("Hasil pengurutan elemen array:");
        for (int i = 0; i < ukuran; i++)
        {
            System.out.print (array [i] + " ");
        }
        System.out.println ();
    }
}

File: Algoritma/Pengurutan/Bubble Sort/Bubble-Sort.py
import random

def main ():
    pilihan = 0
    while pilihan != 3:
        print ("========== BUBBLE SORT ==========")
        print ("1. Isi array secara manual")
        print ("2. Isi array secara otomatis (acak)")
        print ("3. Keluar")
        pilihan = int (input ("Pilih salah satu: "))

        if pilihan == 1:
            ukuran = int (input ("Masukkan batas ukuran array: "))
            array = [int (i) for i in input("Masukkan nilai tiap elemen array:\n").split()]
            urut_dan_tampilkan (ukuran, array)
        elif pilihan == 2:
            ukuran = int (input ("Masukkan batas ukuran array: "))
            batas = int (input ("Masukkan batas angka acak: "))
            array = [random.randint (1, batas) for i in range (ukuran)]
            print ("Elemen array:\n")
            print (*array)
            urut_dan_tampilkan (ukuran, array)
        elif pilihan == 3:
            print ("Terima kasih telah menggunakan program ini!")
        else:
            print ("Input error")

def urut_dan_tampilkan (ukuran, array):
    for i in range (ukuran - 1):
        for j in range (ukuran - i - 1):
            if array [j] > array [j + 1]:
                array [j], array [j + 1] = array [j + 1], array [j]

    print ("Hasil pengurutan elemen array:")
    print (*array)

if __name__ == "__main__":
    main ()